pdesolvers/heat_equation_1d/heat_equation.py

```diff
@@ -0,0 +1,129 @@
+import numpy as np
+from matplotlib import pyplot as plt
+
+from pdesolvers.heat_equation_1d.strategy.solver_strategy import SolverStrategy
+
+
+class HeatEquation:
+
+    def __init__(self, length, x_nodes, time, t_nodes, k, solver: SolverStrategy = None):
```

Comment:
Typically you would have a `Solver` class and you provide a problem to it, rather than a problem class to which you provide a solver. This is how other solvers work (see for example https://alphaville.github.io/optimization-engine/, https://www.cvxpy.org/, etc)